PowerNSX.psm1

```diff
@@ -5186,6 +5190,8 @@ function New-NsxController {
             }
         }
 
+        # Check for presence of optional controller name
+        if ($ControllerName){Add-XmlElement -xmlRoot $ControllerSpec -xmlElementName "name" -xmlElementText $ControllerName.ToString()}
```

Comment:
Because PowerNSX runs in strict mode, this will cause an exception if the user does not specify a controller name as there is no default value for the Param, and therefore the variable wont exist.  Easy fix.  Rather than `if ( $ControllerName)`, we need to test for the existence of the ControllerName parameter in the PSBoundParameters collection as follows `if ( $PSBoundParameters.ContainsKey("ControllerName"))`